### ISSUE TITLE
"Dead" code or undocumented repo type?

### ISSUE THREAD

**Issue by michaelsauter** [opendevstack/ods-jenkins-shared-library#555]
I've come across https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/src/org/ods/orchestration/util/MROPipelineUtil.groovy#L343-L362.

That part is, to my understanding, only executed if the repo is not of type `ods`, `ods-service` or `ods-test` (based on the `if` conditions above). However, according to the docs (https://github.com/opendevstack/ods-jenkins-shared-library/blob/master/docs/modules/jenkins-shared-library/pages/orchestration-pipeline.adoc#configuration) there are only 3 repo types: `ods`, `ods-service` and `ods-test`.

Which makes me wonder if the shell script / Makefile part is dead code and can be removed, or if there is an undocumented component type, or if I am just confused and don't understand the code.

@metmajer do you know more on this?

**Comment by clemensutschig** [comment]
I think it's dead code (as we somewhere default the type to `ods` from what I remember ... )

``` (Project 1003ff)
            // Set repo type, if not provided
            if (!repo.type?.trim()) {
                repo.type = MROPipelineUtil.PipelineConfig.REPO_TYPE_ODS_CODE
            }
```

**Comment by clemensutschig** [comment]
so we should introduce `makefile` and `shellscript` imho

**Comment by metmajer** [comment]
You're right @michaelsauter. This stems from the first executor implementation which happened before the `ods` type got integrated. Then, `ods` became a default as indicated by @clemensutschig which rendered this block a dead code in the end. Believe that there are some use-cases that would benefit from this in one or the other way, but good for me if we bury it in the Git history for now.
